provider/maas/bridgescript_test.go

```diff
@@ -185,6 +197,90 @@ func (s *bridgeConfigSuite) runScript(c *gc.C, pythonBinary, configFile, bridgeP
 	return stdout, result.Code
 }
 
+func (s *bridgeConfigSuite) runScriptWithActivationAndDryRun(c *gc.C, pythonBinary, bridgePrefix string, interfacesToBridge []string) (output []string, exitCode int) {
+	if bridgePrefix != "" {
+		bridgePrefix = fmt.Sprintf("--bridge-prefix=%q", bridgePrefix)
+	}
+
+	script := fmt.Sprintf("%q %q --activate --dryrun %s %s --interfaces-to-bridge=%q",
+		pythonBinary, s.testPythonScript, bridgePrefix, s.testConfigPath, strings.Join(interfacesToBridge, " "))
+	c.Log(script)
+	result, err := exec.RunCommands(exec.RunParams{Commands: script})
+	c.Assert(err, jc.ErrorIsNil, gc.Commentf("script failed unexpectedly"))
+	stdout := strings.Split(string(result.Stdout), "\n")
+	stderr := strings.Split(string(result.Stderr), "\n")
+	output = make([]string, 0, len(stdout)+len(stderr)+1)
+	for _, s := range stdout {
+		s = strings.Trim(s, "\n")
+		if s != "" {
+			output = append(output, s)
+		}
+	}
+	for _, s := range stderr {
+		s = strings.Trim(s, "\n")
+		if s != "" {
+			output = append(output, s)
+		}
+	}
+	return output, result.Code
+}
+
+func (s *bridgeConfigSuite) dryRunExpectedOutputHelper(isBonded, isAlreadyBridged bool, bridgePrefix string, interfacesToBridge []string) []string {
+	output := make([]string, 0)
+	output = append(output, "**** Original configuration")
+	output = append(output, fmt.Sprintf("cat %s", s.testConfigPath))
+	output = append(output, "ifconfig -a")
+	output = append(output, fmt.Sprintf("ifdown --exclude=lo --interfaces=%[1]s $(ifquery --interfaces=%[1]s --exclude=lo --list)", s.testConfigPath))
+	output = append(output, "**** Activating new configuration")
+	if isBonded {
+		output = append(output, "working around https://bugs.launchpad.net/ubuntu/+source/ifenslave/+bug/1269921")
+		output = append(output, "working around https://bugs.launchpad.net/juju-core/+bug/1594855")
+		output = append(output, "sleep 3")
+	}
+	output = append(output, fmt.Sprintf("cat %s", s.testConfigPath))
+	output = append(output, fmt.Sprintf("ifup --exclude=lo --interfaces=%[1]s $(ifquery --interfaces=%[1]s --exclude=lo --list)", s.testConfigPath))
+	output = append(output, "ip link show up")
+	output = append(output, "ifconfig -a")
+	output = append(output, "ip route show")
+	output = append(output, "brctl show")
+	return output
+}
+
+func (s *bridgeConfigSuite) TestBridgeScriptWithoutBondedInterfaceSingle(c *gc.C) {
+	bridgePrefix := "TestBridgeScriptWithoutBondedInterfaceSingle"
+	interfacesToBridge := []string{"eth0"}
+	expectedOutput := s.dryRunExpectedOutputHelper(false, false, bridgePrefix, interfacesToBridge)
```

Comment:
This sort of testing bothers me a little bit, as it feels a bit like a "assert my helper code matches my generating code", rather than just seeing the actual config

s.dryRunExpectedOutput(false, false, bridgePrefix, []string{"eth0"},
 `actual
string of the
new /etc/network/interfaces
`)

Gives people an actual visual indication of what they think they should expect. And a little bit more of a human sanity check.

I've heard other opinions on this, so I wouldn't block on it. But *I* like seeing the concrete string.